### PR TITLE
[wasm] Update Wasm.Build.Tests to build with net6.0

### PR DESCRIPTION
**Pull request by radical**
- This also refactors to tests to be usable for testing with workloads
  in an upcoming PR.
- And skips workload-only tests

### CODE REVIEW

File: src/tests/BuildWasmApps/Wasm.Build.Tests/CommandResult.cs [hunk 1]
```diff
@@ -0,0 +1,42 @@
+using System;
```
Comment (pavelsavara) [inline src/tests/BuildWasmApps/Wasm.Build.Tests/CommandResult.cs:1]: License header ?

**Comment by radical** [comment]
`Libraries Test Run release coreclr Linux x64 Debug failed` - Unrelated - https://github.com/dotnet/runtime/issues/54778

`CoreCLR Pri0 Runtime Tests Run windows arm64 checked` - unrelated, likely - https://github.com/dotnet/runtime/issues/53329

**Comment by radical** [comment]
`Build MacCatalyst x64 Debug AllSubsets_Mono`: timed out after almost 3 hours, though `checks` doesn't seem to be getting the updated status. Unrelated to this PR.

File: src/tests/BuildWasmApps/Wasm.Build.Tests/CommandBase.cs [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+            _label = label;
+        }
+
+        public Command WithWorkingDirectory(string dir)
+        {
+            WorkingDirectory = dir;
+            return this;
+        }
+
+        public Command WithEnvironmentVariable(string key, string value)
```
Comment (lewing) [inline src/tests/BuildWasmApps/Wasm.Build.Tests/CommandBase.cs:43]: make a couple of extension methods for IDictionary<T,T> and use them for the enviroment stuff

File: src/tests/BuildWasmApps/Wasm.Build.Tests/CommandBase.cs [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+            if (extraEnvVars != null)
+            {
+                foreach ((string key, string value) in extraEnvVars)
+                    Environment[key] = value;
+            }
+
+            return this;
+        }
+
+        public virtual CommandResult Execute(string args = "")
```
Comment (lewing) [inline src/tests/BuildWasmApps/Wasm.Build.Tests/CommandBase.cs:60]: The argument construction is a bit confusing, maybe make an Arguments { get; } List<string> and make this. (I have a personal aversion to a single string as args).
```suggestion
        public virtual CommandResult Execute(string[] addtionalArguments)
```

File: src/tests/BuildWasmApps/Wasm.Build.Tests/CommandBase.cs [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+        {
+            _command = command;
+            _label = label;
+        }
+
+        public Command WithWorkingDirectory(string dir)
+        {
+            WorkingDirectory = dir;
+            return this;
+        }
```
Comment (lewing) [inline src/tests/BuildWasmApps/Wasm.Build.Tests/CommandBase.cs:41]: Is this used anywhere?

File: src/tests/BuildWasmApps/Wasm.Build.Tests/CommandBase.cs [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+            {
+                if (e.Data == null)
+                    return;
+
+                output.Add($"[{_label}] {e.Data}");
+                var handler = ErrorDataReceived;
+                if (handler != null)
+                {
+                    handler(s, e);
+                }
```
Comment (lewing) [inline src/tests/BuildWasmApps/Wasm.Build.Tests/CommandBase.cs:97]:
```suggestion
                ErrorDataReceived?.Invoke(s, e);
```

File: src/tests/BuildWasmApps/Wasm.Build.Tests/CommandBase.cs [hunk 6]
```diff
@@ -0,0 +1,200 @@
...
+            {
+                if (e.Data == null)
+                    return;
+
+                output.Add($"[{_label}] {e.Data}");
+                var handler = OutputDataReceived;
+                if (handler != null)
+                {
+                    handler(s, e);
+                }
```
Comment (lewing) [inline src/tests/BuildWasmApps/Wasm.Build.Tests/CommandBase.cs:110]:
```suggestion
                OutputDataReceived?.Invoke(s, e);
```

**Review by lewing — COMMENTED** [review]
Looking good, some thoughts

File: src/tests/BuildWasmApps/Wasm.Build.Tests/CommandBase.cs (hunk 4 above)
Comment (radical) [reply src/tests/BuildWasmApps/Wasm.Build.Tests/CommandBase.cs:41]: not yet :)

File: src/tests/BuildWasmApps/Wasm.Build.Tests/CommandBase.cs (hunk 2 above)
Comment (radical) [reply src/tests/BuildWasmApps/Wasm.Build.Tests/CommandBase.cs]: I want to retain the fluent API.

**Comment by lewing** [comment]
are the runtime pack resolution bits needed in this pr?

**Comment by radical** [comment]
> are the runtime pack resolution bits needed in this pr?

where? the changes in `WasmApp.Native.targets` are needed to make net6.0 test project builds work correctly. Those properties are set elsewhere already.

**Review by lewing — APPROVED** [review]

**Pull request merged**